### PR TITLE
[fix][broker]Close publishLimiter when disable it

### PR DESCRIPTION
**Pull request by AnonHxy**
### Motivation

Close `topicPublishRateLimiter` when it is disabled  to prevent resource leakages

### Modifications

Close `topicPublishRateLimiter` when it change from enable to disable

### Verifying this change

- [x] Make sure that the change passes the CI checks.

  
- [x] `no-need-doc` 
(Please explain why)

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractTopic.java [hunk 1]
```diff
@@ -1166,6 +1166,14 @@ public void updatePublishDispatcher() {
             }
         } else {
             log.info("Disabling publish throttling for {}", this.topic);
+            if (topicPublishRateLimiter != null
+                && topicPublishRateLimiter != PublishRateLimiter.DISABLED_RATE_LIMITER) {
```
Comment (Technoboy-) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractTopic.java:1170]: I think it's no need to test `topicPublishRateLimiter != PublishRateLimiter.DISABLED_RATE_LIMITER`

**Comment by Technoboy-** [comment]
Hi @lhotari 
#10384 has defined `PublishRateLimiter` to extend `AutoCloseable` with `close` signs exception, and cause to catch the exception when updated.  I think we should override the method in this interface to avoid throwing exceptions.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractTopic.java (hunk 1 above)
Comment (AnonHxy) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractTopic.java]: OK, PTAL again @Technoboy-

Comment (Technoboy-) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractTopic.java]: I have override the `close` method by https://github.com/apache/pulsar/pull/15529
So you need to rebase and no need to try/catch.

**Review by Technoboy- — APPROVED** [review]

**Review by Jason918 — APPROVED** [review]

**Pull request merged**